### PR TITLE
Sometimes spot instances fail more than 20 times in a row

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Sometimes the tests are not restarted.

For example: https://github.com/ClickHouse/ClickHouse/pull/44356

This has the only failure of
> PullRequestCI / IntegrationTestsAsan5 (pull_request) Failing after 49m

GitHub Actions due to a spot instance two days ago.
Please note that this prevented us from achieving green CI before New Year.
If there were not these problems with GitHub Actions, Spot instances, and restarts, we'd have green CI in this PR.

**Comment by Felixoid** [comment]
We faced it for the first time since summer. 30 looks like a better option to me.

File: tests/ci/workflow_approve_rerun_lambda/app.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 # Number of times a check can re-run as a whole.
 # It is needed, because we are using AWS "spot" instances, that are terminated often
-MAX_WORKFLOW_RERUN = 20
+MAX_WORKFLOW_RERUN = 100
```
Comment (Felixoid) [inline tests/ci/workflow_approve_rerun_lambda/app.py:28]:
```suggestion
MAX_WORKFLOW_RERUN = 30
```

**Comment by Felixoid** [comment]
The code is deployed

**Pull request merged**